packages/common/src/major2-validation.ts

```diff
@@ -511,10 +538,26 @@ function validateAndRequirement(
     }
     // if there were no solutions added, then there are no valid solutions for the whole AND
     if (solutionsSoFarWithChild.length === 0) {
-      return Err(AndErrorNoSolution(childIdx));
+      const actualIndex = results.findIndex((solution) => {
+        return (
+          solution.type === ResultType.Ok &&
+          solution.ok === childRequirementSolutions
+        );
+      });
+      if (childErrors.length > 0) {
+        return Err(AndErrorUnsatChildAndNoSolution(childErrors, actualIndex));
+      } else {
```

Comment:
Special check for the new combined AndErrorUnsatChildAndNoSolution error (very verbose but also all important information in the name :/).